File: src/dataset/javascript/difference-of-squares_exercism-javascript-ffflorian/gemini/GeminiFlash_difference-of-squares_v1.js
//@ts-check

export class Squares {
  /**
   * @param {number} n
   */
  constructor(n) {
    /** @type {number} */
    this.n = n;
    this._sumOfSquares = (n * (n + 1) * (2 * n + 1)) / 6;
    const sum = (n * (n + 1)) / 2;
    this._squareOfSum = sum * sum;
    this._difference = this._squareOfSum - this._sumOfSquares;
  }

  get sumOfSquares() {
    return this._sumOfSquares;
  }

  get squareOfSum() {
    return this._squareOfSum;
  }

  get difference() {
    return this._difference;
  }
}

File: src/dataset/go/difference_of_squares_Exercism_go_rootulp/gemini/GeminiFlash_difference_of_squares_v1.go
package diffsquares

// SquareOfSum returns (1 + 2 + ... + n) ^ 2
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

// SumOfSquares returns 1^2 + 2^2 + ... + n^2
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// Difference returns the difference between the SquareOfSum(n) and SumOfSquares(n)
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}